packages/data-uri/src/index.ts

```diff
@@ -0,0 +1,81 @@
+import { URL } from 'url';
+
+import { Plugin, RollupError } from 'rollup';
+
+import { dataToEsm } from '@rollup/pluginutils';
+
+const reDataUri = /^([^/]+\/[^;,]+)(;base64)?,([\s\S]*)$/;
+const mimeTypes = {
+  js: 'text/javascript',
+  json: 'application/json'
+};
+
+export default function dataUri(): Plugin {
+  const resolved: { [key: string]: any } = {};
```

Comment:
```suggestion
  const resolved: { [key: string]: { mime?: string, content?: string } } = {};
```

We might want to use a `Map` instead as well.

Reply:
![but why](https://media.giphy.com/media/1M9fmo1WAFVK0/giphy.gif)

not being snarky whatsoever - the only reason I can for see for using Map here is for the sake of using Map. this is never going to be large enough (someone prove me wrong, I dare you) to have perf implications (we're not running in a browser) and there's no immediate benefit of `Map.has()` over `!!object[key]`. New school might say it's more readable but then fundamentals crowd might say they're equal.